charms/k8s/lib/charms/k8s/v0/k8sd_api_manager.py

```diff
@@ -313,8 +414,30 @@ def enable_component(self, name: str, enable: bool):
             enable (bool): True to enable, False to disable the component.
         """
         endpoint = f"/1.0/k8sd/components/{name}"
-        body = {"status": "enable" if enable else "disable"}
-        self._send_request(endpoint, "PUT", body, UpdateComponentResponse)
+        body = {"status": "enabled" if enable else "disabled"}
+        self._send_request(endpoint, "PUT", UpdateComponentResponse, body)
+
+    def is_cluster_bootstrapped(self) -> bool:
+        """Check if K8sd has been bootstrapped.
+
+        Returns:
+            bool: True if the cluster has been bootstrapped, False otherwise.
+        """
+        endpoint = "/1.0/k8sd/cluster"
+        cluster_status = self._send_request(endpoint, "GET", GetClusterStatusResponse)
+        return cluster_status.error_code == 0
```

Comment:
I wonder why we have this extra layer of error codes? Can we not use the HTTP response code?